_posts/2018-03-19-from-python-to-go-to-rust.markdown

```diff
@@ -0,0 +1,80 @@
+
+"Hey, and what about Scala?!". What do you mean by Scala? The better Java? The functional programming language with scalaz? The Object Orienting Programming Functional language that may or may not fail at runtime with a `java.lang.NullPointerException` and needs a 4GB JVM running? I tried it some years ago and definitievely, this is a no go for me.
+
+After discussing with a few people, I decided to give __Go__ a try. It has a __compiler__, __no exceptions__, no `null` (but __null values__) and can handle __concurrency__ nicely.
+
+## Into Go,
+
+I decided to rewrite an internal project that was already done in Python using Go. Just to get a feeling of the differences between the two.
+
+First feeling: learning __Go__ was so easy. In __one evening__, I was able to compile a Proof Of Concept of the projet with basic features developped and some tests written. This was a very pleasant feeling, I was adding features very fast. The compiler messages were helpful, everything was fine.
+
+And at some point, the tragedy started. I needed to add a field to some struct, so I just modified the struct and was ready to analyze the compiler messages to know where this struct was used in order to add the field where it was needed.
+
+I compiled the code and … no error message. Everything went fine. But?! I just added a field to a struct, the compiler should say that my code is not good anymore because I'm not initializing the value where it should be!
+
+The problem is that, not providing a value to a struct is not a problem in __Go__. This value will default to it's [zero value](https://tour.golang.org/basics/12) and everything will compile. This was the __show stopper__ for me. I realized that I __couldn't rely on the compiler__ to get my back when I was doing mistakes. At this point, I was wondering: why should I bother learning __Go__ if the compiler can't do much better than [Python and mypy](http://mypy-lang.org/)? Of course concurrency is much better with __Go__, but the downside of not being able to rely on the compiler was too much for me.
+
+## Into Rust.
+
+So __Go__ was not an option anymore as I realized that what I really needed was a __useful compiler__: a compiler that __should not rely on the fact that I know how to code__ (as it has been proven to be false a lot of times). That's why I took a look at __Rust__.
+
+Rust was not my first choice because it advertises itself as a "system language", and I'm more of a web developer than a system one. But it had some very strong selling points:
+
+- No `null` values but an `Option` type (checked at compile time)
+- No `exceptions` but a `Result` type (checked at compile time)
+- Variables are __immutable__ by default
+- Built for concurrency in mind
```

Comment:
Maybe also list memory safety (and speed), which are the most advertised selling points of the language

Reply:
"Built for concurrency in mind" => "Designed with concurrency in mind"